packages/server/src/auth/utils/cookies.ts

```diff
@@ -0,0 +1,51 @@
+import {
+    SESSION_PREFIX,
+    REFRESH_TOKEN_PREFIX,
+    CrossmintAuthenticationError,
+    type AuthMaterialBasic,
+} from "@crossmint/common-sdk-auth";
+import type { IncomingMessage } from "http";
+import { type GenericRequest, isNodeRequest, isFetchRequest } from "../types/request";
+
+export function getAuthCookies(request: GenericRequest): AuthMaterialBasic {
+    let cookieHeader: string;
+
+    if (isNodeRequest(request)) {
+        cookieHeader = getCookieHeaderFromNodeRequest(request);
+    } else if (isFetchRequest(request)) {
+        cookieHeader = getCookieHeaderFromFetchRequest(request);
+    } else {
+        throw new CrossmintAuthenticationError("Unsupported request type");
+    }
+
+    const { [SESSION_PREFIX]: jwtToken, [REFRESH_TOKEN_PREFIX]: refreshToken } = parseCookieHeader(cookieHeader);
+
+    return { jwtToken, refreshToken };
```

Comment:
when possible, I aim to keeping a method either fully declarative or fully imperative, here it's a bit of both where the first half w/ if-else is imperative, then we have a nice declarative `parseCookieHeader`
I'd be thinking:
```
const cookieHeader = getCookieHeader(request)
const { ... } = parseCookieHeader(...)
return { ... }
```

which has the added benefit of removing some repeated code between `getCookieHeaderFromNodeRequest` and `getCookieHeaderFromFetchRequest` because `getCookieHeader` would have to look like:
```
let cookieHeader;
if (isNodeRequest) {
  cookieHeader = ...
} else if (isFetchRequest) {
  cookieHeader = ...
} 

if (cookieHeader == null) {
  throw
}
return cookieHeader